ote_sdk/ote_sdk/tests/entities/test_task_environment.py

```diff
@@ -0,0 +1,379 @@
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_compare_learning_parameters_learning_rate(self):
+        """
+        <b>Description:</b>
+        Check matches ConfigExample and dummy_config.yaml learning_rate default_value
+
+        <b>Input data:</b>
+        ConfigExample, dummy_config.yaml
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Cheking learning_rate from ConfigExample and dummy_config.yaml
+        """
+
+        env = environment()
+        config_example = env.get_hyper_parameters(ConfigExample)
+
+        env_learning_parameters_learning_rate = \
+            env.get_hyper_parameters().learning_parameters.learning_rate # "default_value"
+        config_example_learning_parameters_learning_rate = \
+            config_example.learning_parameters._default.factory.learning_rate.metadata["default_value"]
+
+        assert env_learning_parameters_learning_rate == 0.05 # From dummy_template.yaml "parameter_overrides"
+        assert config_example_learning_parameters_learning_rate == 0.01 # From ConfigExample
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_compare_learning_parameters_num_checkpoints(self):
+        """
+        <b>Description:</b>
+        Check matches ConfigExample and dummy_config.yaml num_checkpoints default_value
+        "num_checkpoints" is missing in ConfigExample
+
+        <b>Input data:</b>
+        ConfigExample, dummy_config.yaml
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Cheking num_checkpoints from ConfigExample and dummy_config.yaml
+        """
+
+        env = environment()
+        config_example = env.get_hyper_parameters(ConfigExample)
+
+        env_learning_parameters_num_checkpoints = \
+            env.get_hyper_parameters().learning_parameters.num_checkpoints # "default_value"
+
+        # From dummy_config.yaml because it is missing in dummy_template.yaml "parameter_overrides"
+        assert env_learning_parameters_num_checkpoints == 5
+
+        # Attempt to access the missing parameter in ConfigExample
+        with pytest.raises(AttributeError):
+            # AttributeError: type object '__LearningParameters' has no attribute 'num_checkpoints'
+            config_example.learning_parameters._default.factory.num_checkpoints.metadata["default_value"]
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_dummy_config_missing_param(self):
+        """
+        <b>Description:</b>
+        Check dummy_config missing_param
+
+        <b>Input data:</b>
+        dummy_config
+
+        <b>Expected results:</b>
+        Test passes if incoming data is processed correctly
+
+        <b>Steps</b>
+        1. Cheking missing_param dummy_config
```

Comment:
could you find all similar typos and fix?

Reply:
Typos fixed